src/main/scala/mesosphere/marathon/core/task/update/impl/steps/ScaleAppUpdateStepImpl.scala

```diff
@@ -17,12 +19,37 @@ import scala.concurrent.Future
   * Trigger rescale of affected app if a task died or a reserved task timed out.
   */
 class ScaleAppUpdateStepImpl @Inject() (
-    @Named("schedulerActor") schedulerActorProvider: Provider[ActorRef]) extends TaskUpdateStep {
+  @Named("schedulerActor") schedulerActorProvider: Provider[ActorRef])
+    extends TaskUpdateStep with InstanceChangeHandler {
+
   private[this] val log = LoggerFactory.getLogger(getClass)
   private[this] lazy val schedulerActor = schedulerActorProvider.get()
 
   override def name: String = "scaleApp"
 
+  override def process(update: InstanceChange): Future[Done] = {
+    // TODO(PODS): we might want to consider scaling pods in case of status Reserved or Unreachable, which are both
+    // not Terminal. That should be up to a tbd InstanceLostBehavior.-
+    update.status match {
+      case _: InstanceStatus.Terminal =>
+        val runSpecId = update.runSpecId
+        val instanceId = update.id
+        val state = update.status
+        log.info(s"initiating a scale check for runSpec [$runSpecId] due to [$instanceId] $state")
+        // TODO(PODS):
+        //    1) SchedulerActor is awful
+        //    2) yet it handles runSpecs locked due to deployments
+        //    3) if we still use it, we should rename the Message and make the SchedulerActor generic
+        schedulerActor ! ScaleApp(runSpecId)
```

Comment:
I think it works for both, and if not, the easiest thing would be to make it work for both